platform-manageability-agent/internal/comms/comms.go

```diff
@@ -0,0 +1,221 @@
+	var (
+		status  = pb.AMTStatus_DISABLED
+		version string
+	)
+
+	scanner := bufio.NewScanner(strings.NewReader(string(output)))
+	for scanner.Scan() {
+		line := scanner.Text()
+
+		if strings.HasPrefix(line, "Version") {
+			parts := strings.Split(line, ":")
+			if len(parts) > 1 {
+				version = strings.TrimSpace(parts[1])
+				status = pb.AMTStatus_ENABLED
+			}
+		}
+	}
+
+	req := &pb.AMTStatusRequest{
+		HostId:  strings.TrimSpace(string(uuid)),
+		Status:  status,
+		Version: version,
+	}
+	return req, nil
+}
+
+// ReportAMTStatus executes the `rpc amtinfo` command, parses the output, and sends the AMT status to the server.
+func (cli *Client) ReportAMTStatus(ctx context.Context) (pb.AMTStatus, error) {
+	defaultStatus := pb.AMTStatus_DISABLED
+
+	uuid, err := utils.GetSystemUUID()
+	if err != nil {
+		return defaultStatus, fmt.Errorf("failed to retrieve UUID: %w", err)
+	}
+
+	// TODO: RPC location need to be checked.
+	output, err := utils.ExecuteWithRetries("sudo", []string{"./rpc", "amtinfo"})
+	if err != nil {
+		req := &pb.AMTStatusRequest{
+			HostId:  uuid,
+			Status:  defaultStatus,
+			Version: "",
+		}
+		_, reportErr := cli.DMMgrClient.ReportAMTStatus(ctx, req)
+		if reportErr != nil {
+			return defaultStatus, fmt.Errorf("failed to report AMTStatus to DM Manager: %w", reportErr)
+		}
+		return defaultStatus, fmt.Errorf("failed to execute `rpc amtinfo` command: %w", err)
+	}
+
+	req, err := parseAMTInfo(uuid, output)
+	if err != nil {
+		return defaultStatus, fmt.Errorf("failed to parse `rpc amtinfo` output: %w", err)
+	}
+
+	_, err = cli.DMMgrClient.ReportAMTStatus(ctx, req)
+	if err != nil {
+		return defaultStatus, fmt.Errorf("failed to report AMT status: %w", err)
+	}
+	log.Logger.Info("Successfully reported AMT status")
+	return req.Status, nil
+}
+
+// RetrieveActivationDetails retrieves activation details and executes the activation command if required.
+func (cli *Client) RetrieveActivationDetails(ctx context.Context, conf *config.Config) error {
+	uuid, err := utils.GetSystemUUID()
+	if err != nil {
+		return fmt.Errorf("Failed to retrieve activation details: %w", err)
+	}
+
+	req := &pb.ActivationRequest{
+		HostId: uuid,
+	}
+	resp, err := cli.DMMgrClient.RetrieveActivationDetails(ctx, req)
+	if err != nil {
+		return fmt.Errorf("Failed to retrieve activation details: %w", err)
+	}
+
+	log.Logger.Debugf("Retrieved activation details: HostID=%s, Operation=%v, ProfileName=%s",
+		resp.HostId, resp.Operation, resp.ProfileName)
+
+	if resp.Operation == pb.OperationType_ACTIVATE {
+		rpsAddress := fmt.Sprintf("wss://%s/activate", conf.RPSAddress)
+		// TODO:
+		// This is a placeholder, replace with actual logic to fetch the password.
+		// Need to check how to fetch the password from dm-manager, hardcoded for now.
+		password := "P@ssw0rd"
+		cmd := exec.Command("sudo", "rpc", "activate", "-u", rpsAddress, "-n", "-profile", resp.ProfileName, "-password", password)
```

Comment:
this needs to be fixed. All the info should come either from the config file or dm-manager